packages/react-components/react-message-bar-preview/stories/MessageBar/Animation.stories.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import { Button, Link, makeStyles, shorthands, tokens, Field, RadioGroup, Radio } from '@fluentui/react-components';
+import { DismissRegular } from '@fluentui/react-icons';
+import {
+  MessageBar,
+  MessageBarActions,
+  MessageBarTitle,
+  MessageBarBody,
+  MessageBarGroup,
+  MessageBarGroupProps,
+} from '@fluentui/react-message-bar-preview';
+
+const useStyles = makeStyles({
+  container: {
+    display: 'flex',
+    flexDirection: 'column',
+    marginTop: '10px',
+    ...shorthands.gap('10px'),
+    height: '300px',
+    ...shorthands.overflow('auto'),
+    ...shorthands.border('2px', 'solid', tokens.colorBrandForeground1),
+  },
+});
+
+const intents = ['info', 'warning', 'error', 'success'] as const;
+
+interface Entry {
+  intent: (typeof intents)[number];
+  id: number;
+}
+
+export const Animation = () => {
+  const styles = useStyles();
+  const counterRef = React.useRef(0);
+  const [animate, setAnimate] = React.useState<MessageBarGroupProps['animate']>('exit-only');
+  const [messages, setMessages] = React.useState<Entry[]>([]);
+  const prepend = () => {
+    const intentPos = Math.floor(Math.random() * intents.length);
+    const newEntry = {
+      intent: intents[intentPos],
+      visible: true,
+      id: counterRef.current++,
+    };
+
+    setMessages(s => [newEntry, ...s]);
+  };
+
+  const clear = () => {
+    setMessages([]);
+  };
+
+  const dismiss = (id: number) => () => {
+    setMessages(s => {
+      const newState = s.map(entry => ({ ...entry }));
```

Comment:
Is this needed? I think that `.filter()` should be enough now.

Reply:
I can make the change, I was just following the immutability concept